### PR TITLE
ref: Improve serialize data method

### PR DESCRIPTION
**Pull request by raulgomezc**
Closes [Ticket](https://trello.com)

## Testing

Steps to manually verify the change:
1. _Setup_: step to set the environment
2. _Exercise_: How to run the environment and example link: [Test Link](http://127.0.0.1:8000/)
3. _Verify_: what we need to verify

### CODE REVIEW

File: paradise_papers_search/fetch_api/models/Address.py [hunk 1]
```diff
@@ -10,4 +10,17 @@ class Address(DjangoNode):
...
+    @property
+    def serialize(self):
+        return {
+            'properties': {
+                'sourceID': self.sourceID,
+                'country_codes': self.country_codes,
+                'valid_until': self.valid_until,
+                'address': self.address,
+                'countries': self.countries,
+                'node_id': self.node_id,
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/Address.py:22]: Yep, that is it!! Now we are using Neomodel. 👍

File: paradise_papers_search/fetch_api/models/Address.py [hunk 2]
```diff
@@ -10,4 +10,17 @@ class Address(DjangoNode):
     countries     = StringProperty()
     node_id       = StringProperty()
 
+    @property
+    def serialize(self):
+        return {
+            'properties': {
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/Address.py]: Let's call it `node_properties` here and through, please.

File: paradise_papers_search/fetch_api/models/Entity.py [hunk 3]
```diff
@@ -19,7 +19,61 @@ class Entity(DjangoNode):
...
+                'status': self.status,
+            },
+        }
+
+    @property
+    def serialize_connections(self):
+        return [
+            {
+                'nodes_type': 'officer',
+                'nodes_related': [officer.serialize for officer in self.officers.all()],
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/Entity.py]: Excellence work mixing Python list comprehension and Neomodel Object Graph Mapper. 🍰

File: paradise_papers_search/fetch_api/models/Entity.py [hunk 4]
```diff
@@ -19,7 +19,61 @@ class Entity(DjangoNode):
...
+                'nodes_related': [entity.serialize for entity in self._entity_relationships()],
+            },
+        ]
+
+    def _entity_relationships(self):
+        results         = self.cypher("START p=node({self}) MATCH n=(p)<-[r]->(x:Entity) RETURN r, x.node_id as Node_id")
+        relationships   = []
+
+        for row in results[0]:
+            relationships.append(self.nodes.get(node_id=row[1]))
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/Entity.py]: Good work getting the node through Neomodel, now you can take advantage of the serialize methods.

File: paradise_papers_search/fetch_api/models/Entity.py [hunk 5]
```diff
@@ -19,7 +19,61 @@ class Entity(DjangoNode):
...
+                'nodes_related': [other.serialize for other in self.others.all()],
+            },
+            {
+                'nodes_type': 'entity',
+                'nodes_related': [entity.serialize for entity in self._entity_relationships()],
+            },
+        ]
+
+    def _entity_relationships(self):
+        results         = self.cypher("START p=node({self}) MATCH n=(p)<-[r]->(x:Entity) RETURN r, x.node_id as Node_id")
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/Entity.py]: I assume we can not fully express this query with Neomodel OGM and that is why we are using Cypher. ⚠️

File: paradise_papers_search/fetch_api/models/helpers.py [hunk 6]
```diff
@@ -15,65 +15,6 @@
...
-    'ibcRUC',
-    'name',
-    'country_codes',
-    'incorporation_date',
-    'node_id',
-    'status',
-]
-
-# Serialize data to be JSON parsable
-### TODO: Refactor and impplement serializers on models
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/helpers.py:35]: yeah, this was really bad 🤕

File: paradise_papers_search/fetch_api/views.py [hunk 7]
```diff
@@ -1,7 +1,7 @@
 from rest_framework.views import APIView
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/views.py:1]: I see we are using this rest_framework... are we sure we need it? I mean if there is a way to do anything with Django we should use Django, if not this is okay. ⚠️

**Review by junior1q94 — CHANGES_REQUESTED** [review]

File: paradise_papers_search/fetch_api/models/Entity.py (hunk 5 above)
Comment (raulgomezc) [reply paradise_papers_search/fetch_api/models/Entity.py]: This function already existed but it was returning a tuple, I just modify it so I could use list comprehension on it after

File: paradise_papers_search/fetch_api/views.py (hunk 7 above)
Comment (raulgomezc) [reply paradise_papers_search/fetch_api/views.py:1]: The framework is necessary for the type of responses we are using and it is specially designed for Django

File: paradise_papers_search/fetch_api/models/Entity.py [hunk 8]
```diff
@@ -19,7 +19,61 @@ class Entity(DjangoNode):
...
+        relationships   = []
+
+        for row in results[0]:
+            relationships.append(self.nodes.get(node_id=row[1]))
```
Comment (mariamera) [inline paradise_papers_search/fetch_api/models/Entity.py]: Hii , question. Why here you are just appending the node? we also need the type of relationship between nodes. Are you getting them somewhere else? thanks

File: paradise_papers_search/fetch_api/models/helpers.py (hunk 6 above)
Comment (mariamera) [reply paradise_papers_search/fetch_api/models/helpers.py:35]: agree D: 🌮

File: paradise_papers_search/fetch_api/models/helpers.py [hunk 9]
```diff
@@ -90,14 +31,14 @@ def filter_nodes(node_type, name, country, jurisdiction):
...
-    count_node          = len(node)
-
-    return count_node
+    count = {}
+    search_word             = count_info['name']
+    country                 = count_info['country']
+    is_jurisdiction         = count_info['jurisdiction']
+    for node_type in list(MODEL_ENTITIES):
+        node                = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, is_jurisdiction)
+        count[node_type]    = len(node)
```
Comment (mariamera) [inline paradise_papers_search/fetch_api/models/helpers.py]: 👍

File: paradise_papers_search/fetch_api/models/Officer.py [hunk 10]
```diff
@@ -12,4 +12,30 @@ class Officer(DjangoNode):
...
+    def serialize_connections(self):
+        return [
+            {
+                'nodes_type': 'address',
+                'nodes_related': [address.serialize for address in self.addresses.all()],
+            },
+            {
+                'nodes_type': 'entity',
+                'nodes_related': [entity.serialize for entity in self.entities.all()],
+            },
```
Comment (mariamera) [inline paradise_papers_search/fetch_api/models/Officer.py]: note: still need intermediaries  and  others  relationships. and a method like ` _entity_relationships` but for `Officer`.

File: paradise_papers_search/fetch_api/views.py [hunk 11]
```diff
@@ -10,7 +10,7 @@ def get(self, request):
             'country': request.GET.get('c', ''),
             'jurisdiction': bool(request.GET.get('j', False)),
```
Comment (mariamera) [inline paradise_papers_search/fetch_api/views.py]: Why jurisdiction is a bool and country isn't ? The way they work are pretty much the same.   Thanks

**Review by mariamera — CHANGES_REQUESTED** [review]
👍 Amazing work

**Review by oscarpolanco — APPROVED** [review]
LGTM after address the comments
good job!!!

File: paradise_papers_search/fetch_api/models/helpers.py [hunk 12]
```diff
@@ -90,14 +31,14 @@ def filter_nodes(node_type, name, country, jurisdiction):
...
-    node                = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, is_jurisdiction)
-    count_node          = len(node)
-
-    return count_node
+    count = {}
+    search_word             = count_info['name']
+    country                 = count_info['country']
+    is_jurisdiction         = count_info['jurisdiction']
+    for node_type in list(MODEL_ENTITIES):
+        node                = filter_nodes(MODEL_ENTITIES[node_type], search_word, country, is_jurisdiction)
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/helpers.py]: Please, call it nodeSet here and through. To make it clearer.

File: paradise_papers_search/fetch_api/models/helpers.py [hunk 13]
```diff
@@ -111,48 +52,15 @@ def fetch_nodes(fetch_info):
...
 
-    for row in results[0]:
-        list.append((row[0].type, self.nodes.get(node_id=row[1])))
-
-    return list
 
 def fetch_node_details(node_info):
     node_type = node_info['node_type']
     node_id = node_info['node_id']
     node = MODEL_ENTITIES[node_type].nodes.get(node_id=node_id)
```
Comment (junior1q94) [inline paradise_papers_search/fetch_api/models/helpers.py]: Please, call it nodeSet here and through. To make it clearer.

**Review by mariamera — APPROVED** [review]

**Review by junior1q94 — APPROVED** [review]

**Pull request merged**